frontend/src/app/main/ui/workspace/tokens/modals/themes.cljs

```diff
@@ -384,41 +387,51 @@
            (st/emit! (wdt/update-token-theme [(:group theme) (:name theme)] theme'))))]
 
     [:> edit-create-theme*
-     {:set-state set-state
+     {:state state
+      :change-view change-view
       :theme theme
       :on-save on-save
       :is-editing true}]))
 
 (mf/defc create-theme
-  [{:keys [set-state]}]
+  [{:keys [state change-view]}]
   (let [theme (ctob/make-token-theme :name "")
-
         on-save
         (mf/use-fn
          (fn [theme]
            (st/emit! (ptk/event ::ev/event {::ev/name "create-tokens-theme"})
                      (wdt/create-token-theme theme))))]
 
     [:> edit-create-theme*
-     {:set-state set-state
+     {:state state
+      :change-view change-view
       :theme theme
       :on-save on-save}]))
 
 (mf/defc themes-modal-body*
   {::mf/private true}
   []
-  (let [themes    (mf/deref refs/workspace-token-themes-no-hidden)
-        state     (mf/use-state #(if (empty? themes)
-                                   {:type :create-theme}
-                                   {:type :themes-overview}))
-        set-state (mf/use-fn #(swap! state %))
-        component (case (:type @state)
+  (let [themes      (mf/deref refs/workspace-token-themes-no-hidden)
+        state*      (mf/use-state #(if (empty? themes)
+                                     {:type :create-theme}
+                                     {:type :themes-overview}))
+        state       (deref state*)
+
+        change-view (mf/use-fn
+                     (fn [type & [state']]
+                       (swap! state* (fn [current-state]
+                                       (merge
+                                        {:type type
+                                         :prev-type (:type current-state)}
+                                        state')))))
```

Comment:
This is a bit subtle... I don't like the idea that sub components need to know the structure of the state. Instead I'd prefer something like this:

```clojure
        change-view (mf/use-fn
                     (fn [type & {:keys [:theme-path]}]
                       (swap! state* (fn [current-state]
                                       (cond-> current-state
                                         :always
                                         (assoc :type type)

                                         (= type :edit-theme)
                                         (assoc :theme-path theme-path))))))
```

What do you think? I consider this to be more explicit, and to introduce less coupling between this component and the inner ones.